### DISCUSSION TITLE
Annoyance: two windows when opening EPUB from File Explorer in Windows

### DISCUSSION THREAD

**Discussion by PhilippeBruno** [edrlab/thorium-reader#1815] [Q&A]
Whenever we open EPUBs from File Explorer in Windows, two windows open. This creates an annoying situation where two windows must be closed to effectively close Thorium.

**Comment by panaC** [comment]
This commit https://github.com/edrlab/thorium-reader/commit/5cb14e7a906e3e4a2577d71f6a5070443e910233 fixed the command line to open multiple files from the file explorer.

The reader window ask to detach the library and reader views because at this point there are no informations about the number of publications that will be opened. It's an event callable any time during the app lifecycle.
https://github.com/edrlab/thorium-reader/commit/5cb14e7a906e3e4a2577d71f6a5070443e910233#diff-e372a717421f70737dd4d6b73af739846902fd3296a898754c66ad626b7ccccfR48

I understand your request. And we will discuss it with the core team asap.

**Comment by danielweck** [comment]
@panaC is this okay to aim to fix this issue in the Thorium 2.2 project?

**Comment by panaC** [comment]
technically is very difficult to fixed this.

culprit : RACE CONDITION

**Comment by panaC** [comment]
It's more safe to run in detach mode the reader windows and the library windows from the main process.

**Comment by panaC** [comment]
My first advice can i give you is to let open a library window in back, in this case you do not have the issue to close 2 windows but only one. It's an 'hacky' solution, but I thinks in works pretty well. On mac-os we doesn't have this kind of issue because a main process stay in background idle.

**Comment by panaC** [comment]
It's my humble opinion maybe @danielweck do you have a better idea how to fix that ?

**Comment by PhilippeBruno** [comment]
@panaC, thanks for your effort. Perhaps not the solution I was expecting, but nevertheless a solution.
